### PR TITLE
website blocker script

### LINKED ISSUE #1129: Website Blocker
# Aim

- Block a website

# Details

- Used to block all websites given as a list in the python script

### PR DESCRIPTION
**Pull request by aanshi18**
# Description

Website Blocker Python Script

## Fixes #1129 

Used to block all websites given as a list in the python script

## Type of change

- [.] New feature (non-breaking change which adds functionality)

## Checklist:

- [.] My code follows the style guidelines(Clean Code) of this project
- [.] I have performed a self-review of my own code
- [.] I have commented my code, particularly in hard-to-understand areas
- [.] I have created a helpful and easy to understand `README.md`
- [.] My documentation follows [`Template for README.md`](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/Template%20for%20README.md)
- [.] I have added the project meta data in the PR template.
- [.] I have created the ``requirements.txt`` file if needed.

## Project Metadata

`` If there is no-file/nothing to fill the below fields with, then type: none ``

`` Example: `` If no requirements.txt needed/present, then type none in ``Requirments``.

Category:
- [ ] Calculators
- [ ] AI/ML
- [ ] Scrappers
- [ ] Social_Media
- [ ] PDF
- [ ] Image_Processing
- [ ] Video_Processing
- [ ] Games
- [ ] Networking
- [x] OS_Utilities
- [ ] Automation
- [ ] Cryptography
- [ ] Computer_Vision
- [ ] Fun
- [ ] Others

Title: Website Blocker

Folder: Website-Blocker

Requirements: none

Script: website_blocker.py

Arguments: none

Contributor: aanshi18

Description: Used to block all websites given as a list in the python script

### CODE REVIEW

**Comment by santushtisharma10** [comment]
@kaustubhgupta i am unable to run the script on my local system, please check its working

**Comment by kaustubhgupta** [comment]
> @kaustubhgupta i am unable to run the script on my local system, please check its working

sure

File: Website-Blocker/website_blocker.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# importing the required libraries
+from datetime import datetime as dt
+
+# Change the list to the websites you wish to block
+websites = [ "www.youtube.com", "twitter.com","www.skype.com/en", "www.facebook.com","www.instagram.com"]
```
Comment (kaustubhgupta) [inline Website-Blocker/website_blocker.py:5]: take the websites input from user

File: Website-Blocker/website_blocker.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+# Change the path to your hosts files folder
+host = "C:\Windows\System32\drivers\etc\hosts"
+# change the path to where you want to redirect 
+redirect = "127.0.0.1:8080"
+
+# Change the date when you want to start and end
+start = dt(2021, 4, 25)
+end = dt(2021, 4, 24)
```
Comment (kaustubhgupta) [inline Website-Blocker/website_blocker.py:13]: take this input from user

File: Website-Blocker/website_blocker.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        with open(host, "r+") as file:
+            temp = file.readlines()
+            file.seek(0)
+            for line in temp:
+                if not any(site in line for site in websites):
+                    file.write(line)
+            # truncate command deletes the sites to be blocked from host file
+            file.truncate()
+        print("The site has been unblocked.")
+        break
```
Comment (kaustubhgupta) [inline Website-Blocker/website_blocker.py:39]: how the logic is working? how the user is removing websites from blocked hosts?

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]

**Comment by avinashkranjan** [comment]
Hey @aanshi18  are you going to update the PR..or Should I close it?

**Comment by aanshi18** [comment]
Yes, I will update it within a day.

**Comment by aanshi18** [comment]
I have updated the script.

File: Website-Blocker/website_blocker.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+else:
+    hosts_path = "/etc/hosts"
+
+redirect_ip = "127.0.0.1"
+
+website_list = []
+
+# Input of the hours for blocking websites
+print("Enter the hour time(24hr format, Ex.16):")
+start_time = int(input("Enter the starting time:"))
```
Comment (kaustubhgupta) [inline Website-Blocker/website_blocker.py:18]:
```suggestion
start_time = int(input("Enter the starting time: "))
```

File: Website-Blocker/website_blocker.py [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+end_time = int(input("Enter the endin time: "))
```
Comment (kaustubhgupta) [inline Website-Blocker/website_blocker.py:19]:
```suggestion
end_time = int(input("Enter the ending time: "))
```

File: Website-Blocker/website_blocker.py [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+                if website in content:
+                    pass
+                else:
+                    file.write(f"{redirect_ip} {website}\n")
+    else:
+        with open(hosts_path, 'r+') as file:
+            content = file.readlines()
+            file.seek(0)
+            for line in content:
+                if not any(website in line for website in website_list):
```
Comment (kaustubhgupta) [inline Website-Blocker/website_blocker.py:52]: include the logic to unblock the websites

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]

**Pull request closed without merging**